mlflow/spark.py

```diff
@@ -98,10 +99,20 @@ def get_default_pip_requirements():
     if Version(pyspark.__version__) < Version("3.4"):
         # Versions of PySpark < 3.4 are incompatible with pandas >= 2
         reqs.append("pandas<2")
+
+    if is_spark_connect_model:
+        reqs.extend([
+            # Spark connect Model uses spark torch distributor to train model
```

Comment:
Do we have docs for spark connect models?

Reply:
spark 3.5 haven't been released, you have to check API docs in source code for now.

Reply:
To summarize, algorithms in `pyspark.ml.connect` have very similar interfaces with `pyspark.ml`, the major differences between their APIs are:

- spark ml-connect models supports saving to / loading from local filesystem path (model.saveToLocal / model.loadFromLocal)
- spark ml-connect models supports running inference on local pandas dataframe, i.e., we can run `model.transform(pandas_dataframe)` and it appends prediction result columns into the input pandas_dataframe.

Reply:
Do we have a 3.5 release candidate?

Reply:
We can use this https://dist.apache.org/repos/dist/dev/spark/v3.5.0-rc4-bin/pyspark-3.5.0.tar.gz

Reply:
@HyukjinKwon Can we upload RC versions on PyPI (e.g. as `3.5.0.rc4`)?

Reply:
In theory, we can do according to [PEP 440](https://peps.python.org/pep-0440/#pre-releases) but we have not done that so far ..

Reply:
@HyukjinKwon Got it, nothing urgent at all. Just a nice-to-have for us to make it easier to run tests against new spark versions.